packages/dotcom-ui-header/src/components/sub-navigation/partials.tsx

```diff
@@ -73,9 +73,10 @@ const SubSections = ({ items }: { items: TNavMenuItem[] }) => {
       data-trackable="subsections">
       {items.map((item, index) => {
         const selected = item.selected ? 'o-header__subnav-link--highlight' : ''
+        const subnavItemId = { 'data-id': item.id ? `subnav-${item.id}` : null }
```

Comment:
It is possible to add the `data-id` property to the object only if the value upon which it depends exists, rather than adding it by default (and setting its value to `null` when the dependent value does not exist).

```
const listItemProps = {
   ...(item.id && { 'data-id': `subnav-${item.id}` })
}
```

And then:
 ``<li className="o-header__subnav-item" key={`item-${index}`} { ...listItemProps }>``

This will also mean we do not need to add these instances of `data-id={null}` to `packages/dotcom-ui-header/src/__test__/components/__snapshots__/MainHeader.spec.tsx.snap`.

It might be worth adding a test to ensure that the correct `data-id` value is output when an `item.id` value is provided (bearing in mind the opinions devs shared about snapshot tests in the last dev huddle).

Reply:
I had this same concern this morning but it looks like React won't output the attribute if it's set to `null`, (jest builds the components differently) so we should be safe with this. We've used the same pattern in a few other places in Page Kit so it's internally consistent.

Reply:
Okay, so it's a toss-up between:

- Write code that is built by our build process in line with how Jest builds, meaning that what we test in Jest is more reliably a reflection of what we build in production.
- Keep the pattern consistent throughout Page Kit.

I think there is a strong case for both so will leave the decision to you.